src/parsing/literalParser.ts

```diff
@@ -162,10 +163,12 @@ export const elementContentChar = preceded(
 export const cdataSection: Parser<IAST> = map(
 	delimited(
 		tokens.CDATA_OPEN,
-		star(
-			preceded(
-				peek(not(tokens.CDATA_CLOSE, ['CDataSection content may not contain "]]>"'])),
-				char
+		cut(
```

Comment:
`delimited` has a pretty handy fourth argument to automatically insert a `cut` after the opening parser succeeds. Setting that to `true` instead of wrapping the middle parser would also mark failures in the closing delimiter be considered fatal.